Xamarin.Forms.Core/Shell/ShellItemCollection.cs

```diff
@@ -14,7 +30,103 @@ internal sealed class ShellItemCollection : IList<ShellItem>, INotifyCollectionC
 
 		public int Count => Inner.Count;
 		public bool IsReadOnly => ((IList<ShellItem>)Inner).IsReadOnly;
-		internal IList<ShellItem> Inner { get; set; }
+		internal IList<ShellItem> Inner
+		{
+			get
+			{
+				return _inner;
+			}
+			set
+			{			
+				_inner = value;
+				((INotifyCollectionChanged)_inner).CollectionChanged += InnerCollectionChanged;
```

Comment:
Shouldn't we unsubscribe from this? On dispose or if we pass a new "inner" ?

Reply:
This list should never change. It's a readonly BP
```C#
static readonly BindablePropertyKey ItemsPropertyKey = BindableProperty.CreateReadOnly(nameof(Items), typeof(ShellItemCollection), 
```